README.rst

```diff
@@ -292,14 +292,35 @@ To add metadata to a table:
 
 .. code-block:: python
 
-    table = Table('mytable', ..., bigquery_description='my table description', bigquery_friendly_name='my table friendly name')
+    table = Table('mytable', ...,
+        bigquery_description='my table description',
+        bigquery_friendly_name='my table friendly name',
+        bigquery_default_rounding_mode="ROUND_HALF_EVEN",
+        bigquery_expiration_timestamp=datetime.datetime.fromisoformat("2038-01-01T00:00:00+00:00"),
+    )
 
 To add metadata to a column:
 
 .. code-block:: python
 
     Column('mycolumn', doc='my column description')
 
+To create a clustered table:
+
+.. code-block:: python
+
+    table = Table('mytable', ..., bigquery_clustering_fields=["a", "b", "c"])
+
+To create a partitioned table:
+
+.. code-block:: python
+
+    table = Table('mytable', ...,
+        bigquery_partitioning="DATE(mytimestamp)",
```

Comment:
Good question! My first thought was naming this option [partition_expression](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#partition_expression) as done in the DDL statements reference.

Then, I tried to get some inspiration from the python SDK options that use a different approach and distinguish [time_partitioning](https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.table.Table#google_cloud_bigquery_table_Table_time_partitioning) from [range_partitioning](https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.table.Table#google_cloud_bigquery_table_Table_range_partitioning ) but did not want to stick to the SDK interface to set the partitioning expression that can be much more simply described.

I finally choose `partitioning` option name which is the base name used in many options and conveys well to the purposes.

Please share your thoughts, and let me know if you believe any changes are necessary.

Reply:
My understanding is that the convention used by sqlalchemy is for dialects to prepend the name of the dialect before an argument that is solely present in that dialect:

```
* bigquery_partition_expiration_days
* bigquery_require_partition_filter
* postgresql_ignore_search_path
* sqlite_on_conflict
```
It is also my understanding that because we are a client and are interfacing with the BigQuery API, that we aim to mirror the API names in particular ways to make maintenance easier, reduce cognitive load, etc.

So in this case... we would want to use one of the existing partitioning parameters (e.g. `range_partitioning`, `time_partitioning`) instead of attempting to create our own (see [example here](https://cloud.google.com/bigquery/docs/creating-partitioned-tables#create_an_integer-range_partitioned_table) of the parameter being set in `python-bigquery` directly. Notice the relationship between the `range_partitioning` attribute and the `RangePartitioning` class).

```
table = bigquery.Table(table_id, schema=schema)
table.range_partitioning = bigquery.RangePartitioning(
    field="zipcode",
    range_=bigquery.PartitionRange(start=0, end=100000, interval=10),
)
```

Reply:
Yes, I understand this. To be honest, I precisely tried to avoid this as I found the partitioning API of the Python SDK bloated vs the simple `partitioning_expression` from the DDL statements reference.

Anyway, I understand that we should not increase the mental overhead and keep consistency between APIs.

As we already depend on the [google-cloud-bigquery](https://github.com/googleapis/python-bigquery) package I will use the same  `RangePartitioning` and `TimePartitioning` classes to describe the partitioning expression.

Reply:
@chalmerlowe 

I'm finally pretty happy with the results. Let me know what you think!
https://github.com/googleapis/python-bigquery-sqlalchemy/blob/1944f95afd843dad2c8d4a2808e031a1f3fdb2ac/tests/unit/test_table_options.py#L170-L188

We could plan to implement `RangePartitiong` in a next iteration. This PR is big enough, and I'd like to see it merged not too late :slightly_smiling_face:.